test/aws/gather_logs.yml

```diff
@@ -0,0 +1,58 @@
+---
+- name: Set path to artifact dir
+  set_fact:
+    scaleup_artifacts_dir: /tmp/artifacts/scaleup
+
+- name: Create log directory
+  local_action:
+    module: file 
+    path: "{{ scaleup_artifacts_dir }}" 
+    state: directory
+
+- name: Collect a list of containers
+  command: crictl ps -a -q
+  register: crictl_ps_output
+  ignore_errors: true
+- name: Collect container logs
+  command: "crictl logs {{ item }}"
+  register: crictl_logs_output
+  with_items: "{{ crictl_ps_output.stdout_lines }}"
+  ignore_errors: true
+- name: Write container logs locally
+  local_action: 
+    module: copy 
+    content: "{{ crictl_logs_output }}"
+    dest: "{{ scaleup_artifacts_dir }}/{{ inventory_hostname }}-containers.log"
+
+- name: Get crio logs
+  command: journalctl --no-pager -u cri-o
+  register: crio_logs
+  ignore_errors: true
+- name: Write crio logs locally
+  local_action: 
+    module: copy 
+    content: "{{ crio_logs }}"
+    dest: "{{ scaleup_artifacts_dir }}/{{ inventory_hostname }}-crio.log"
+  ignore_errors: true
+
+- name: Get kubelet logs
+  command: journalctl --no-pager -u kubelet
+  register: kubelet_logs
+  ignore_errors: true
+- name: Write kubelet logs locally
+  local_action: 
+    module: copy 
+    content: "{{ kubelet_logs }}"
+    dest: "{{ scaleup_artifacts_dir }}/{{ inventory_hostname }}-kubelet.log"
+  ignore_errors: true
+
+- name: Get filenames
+  local_action: command ls {{ scaleup_artifacts_dir }}
+  register: log_files
+
+- name: Archive Log Files
+  local_action:
+    module: archive
+    path: "{{ scaleup_artifacts_dir }}/{{ item }}"
+    remove: yes
```

Comment:
This seems to be causing `msg: 'Unable to remove source file: [Errno 2] No such file or directory: ''/tmp/artifacts/scaleup/10.0.143.129-containers.log'''`

Reply:
I am currently trying to debug this. If I just remove `remove: yes` then it defeats the whole purpose of archiving. I could just forget the archiving step. These logs will only be collected on failures, so I'm not sure if the disk usage is a concern. I am running a playbook now with some debugging info added.

I'm also trying to debug another couple of issues: 
* for some reason the containers logs are failing with `{"msg": "'dict object' has no attribute 'stdout_lines'", "failed": true}` but I can't figure out why. The `dict_object` is `crictl_ps_output`
* The journalctl logs seem to return `--no entries--` that seems wrong to me.

I am trying to work through these issues today.

Reply:
perhaps it need `become: true` here?